framework/src/media/Decoder.cpp

```diff
@@ -71,24 +71,38 @@ bool Decoder::getFrame(unsigned char* buf, size_t* size, unsigned int* sampleRat
 {
 #ifdef CONFIG_AUDIO_CODEC
 	if (mPlayer.audio_type == 0) {
-		mPlayer.audio_type = _get_audio_type(mPlayer.rbsp);
-		medvdbg("audio_type %d\n", mPlayer.audio_type);
+		mPlayer.audio_type = pv_player_get_audio_type(&mPlayer);
 
-		if (_init_decoder(&mPlayer) != 0) {
-			meddbg("Error! _init_decoder failed!\n");
-			return false;
+		if(mPlayer.audio_type) {
+			if (pv_player_init_decoder(&mPlayer, mPlayer.audio_type) != 0) {
```

Comment:
Here, player will call rb_init, and allocate buffer through the calloc.
But getFrames calling every time, I think this will bring memory leak.